bigquery/google/cloud/bigquery/dataset.py

```diff
@@ -26,6 +27,9 @@
 from google.cloud.bigquery.table import TableReference
 
 
+_PROJECT_PREFIX_PATTERN = re.compile(r"(?P<project_id>\S+\:\S+)\.+(?P<dataset_id>\S+)$")
```

Comment:
I think this will match patterns with too many `.` characters. Let's try something like:

```
(?P<project_id>\S+\:[^.]+)\.(?P<dataset_id>[^.]+)$
```

Reply:
I see what you mean, sorry for misunderstanding. Appreciate your help.